docs/hooks.md

```diff
@@ -28,5 +28,43 @@ from snapd (or need to provide information to snapd) they can utilize the
 
 ## Supported Hooks
 
-**Note:** The development of specific hooks is ongoing. None are currently
-supported.
+**Note:** The development of specific hooks is ongoing.
+
+
+### `configure`
+
+The `configure` hook will be called whenever the user requests a configuration
+change via the `snap set` command. The hook should use `snapctl get` to retrieve
+the requested configuration from snapd, and act upon it. If it exits non-zero,
+the configuration will not be applied.
+
+
+#### `configure` example
+
+Say the user runs:
+
+```bash
+$ snap set <snapname> username=foo password=bar
+```
+
+The `configure` hook would be located within the snap at `meta/hooks/configure`.
+An example of what it might contain is:
+
+```bash
+#!/bin/sh
+
+if ! username=$(snapctl get username); then
+    echo "Username is required"
+    exit 1
+fi
+
+if ! password=$(snapctl get password); then
+    echo "Password is required"
+    exit 1
+fi
+
+# Handle username and password, perhaps write to a credential file of some sort.
+echo "user=$username" > $SNAP_DATA/credentials
+echo "password=$password" >> $SNAP_DATA/credentials
+chmod 600 $SNAP_DATA/credentials
```

Comment:
On the next edit of this file which should come once we have "snapctl error", let's please either drop the chmod, or move it up before the password is written out.